src/main/java/com/snowplowanalytics/snowplow/tracker/payload/TrackerPayload.java

```diff
@@ -40,11 +40,11 @@
     @Override
     public void add(final String key, final String value) {
         if (key == null || key.isEmpty()) {
-            LOGGER.error("Invalid key detected: {}", key);
+            LOGGER.error("Null or empty key detected");
```

Comment:
Is there a reason why this one is error and the one below is info?

Reply:
Yeah, I pondered that when I looked at it. If I had to guess, an invalid key is utterly useless (you can't add a null key to a JSON `{ null: "something"}` makes no sense). However... you can have empty or null values (`{key: null, key2: ""}`) but we strip them out in the trackers to reduce payload size (since atomic event values get stringified in the pipeline anyway).

Reply:
We could remove the `info` altogether but it's informative to tell the user they are trying to capture nulls or empty strings when there is no need to (that's maybe some processing you don't need to do), so I think it's ok as it is.